### PR TITLE
[REF] web: convert signature field/widget to owl

### PR DESCRIPTION
**Pull request by FrancoisGe**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/web/static/src/views/fields/signature/name_and_signature.js [hunk 1]
```diff
@@ -0,0 +1,358 @@
...
+import { renderToString } from "@web/core/utils/render";
+import { getDataURLFromFile } from "@web/views/fields/file_handler";
+
+const { Component, useState, onWillStart, useRef, useEffect } = owl;
+
+export class NameAndSignature extends Component {
+    setup() {
+        this.rpc = useService("rpc");
+
+        this.htmlId = _.uniqueId();
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/signature/name_and_signature.js:17]: no _

File: addons/web/static/src/views/fields/signature/name_and_signature.js [hunk 2]
```diff
@@ -0,0 +1,358 @@
...
+        this.defaultName = this.props.signature.name || "";
+        this.defaultFont = this.props.defaultFont || "";
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/signature/name_and_signature.js:19]: use defaultProps instead

File: addons/web/static/src/views/fields/signature/name_and_signature.js [hunk 3]
```diff
@@ -0,0 +1,358 @@
...
+        });
+
+        const signature = useRef("signature");
+        useEffect(
+            (el) => {
+                if (el) {
+                    this.$signatureField = $(".o_web_sign_signature");
+                    this.$signatureField.on("change", () => {
+                        this.props.signature.isSignatureEmpty = this.isSignatureEmpty;
+                    });
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/signature/name_and_signature.js:57]: you don't need jquery for this

File: addons/web/static/src/views/fields/signature/name_and_signature.js [hunk 4]
```diff
@@ -0,0 +1,358 @@
...
+        );
+    }
+
+    /**
+     * Draws the current name with the current font in the signature field.
+     */
+    drawCurrentName() {
+        const font = this.fonts[this.currentFont];
+        const text = this.getCleanedName();
+        const canvas = this.$signatureField.find("canvas")[0];
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/signature/name_and_signature.js:77]: would be nice not to have jquery usage all over the place (you probably only need it where you call jSignatue)

File: addons/web/static/src/views/fields/signature/name_and_signature.js [hunk 5]
```diff
@@ -0,0 +1,358 @@
...
+                color: this.fontColor,
+            })
+        );
+        $svg.attr({
+            xmlns: "http://www.w3.org/2000/svg",
+            "xmlns:xlink": "http://www.w3.org/1999/xlink",
+        });
+
+        return "data:image/svg+xml," + encodeURI($svg[0].outerHTML);
+    }
```
Comment (aab-odoo) [inline addons/web/static/src/views/fields/signature/name_and_signature.js:139]: no jquery here

File: addons/web/static/src/views/fields/signature/name_and_signature.js (hunk 3 above)
Comment (FrancoisGe) [reply addons/web/static/src/views/fields/signature/name_and_signature.js:57]: Yes I need it, for the event create by JSsignature. It doesn't work with a basic listener

**Pull request closed without merging**